app/components/collections/show/header_component.html.erb

```diff
@@ -1,12 +1,12 @@
 <div class="d-flex justify-content-between">
   <div>
     <%= render Elements::HeadingComponent.new(level: :h1, text: title, classes: 'mb-4 d-inline-block') %>
-    <%= render Edit::EditIconButtonComponent.new(presenter:, classes: 'mx-2 py-0 mb-2', icon_classes: 'h4') %>
+    <%= render Edit::EditIconButtonComponent.new(presenter:, classes: 'mx-2 py-0 mb-2', icon_classes: 'h4') if helpers.allowed_to?(:edit?, presenter.collection) %>
```

Comment:
Putting these `if...allowed_to?` here made the most sense. I tried a couple of different patterns for using the component `render?` method but it felt pretty convoluted passing the results of `allowed_to?` through the controller/presenter. If we'd prefer a pattern like that I can do that as well, but this seems the most straightforward.

Reply:
Minor, but I would put the `helpers.allowed_to?(:edit?, presenter.collection)` logic in the class part of the component in a method called `edit?`. It keeps logic out of the erb and makes it easier to read.

Reply:
That's a good idea, I had gone a bit further than that and it felt convoluted, but this feels more like what I was hoping for. :+1: